### PR TITLE
Add generic API call logs

### PR DESCRIPTION
**Pull request by elizabethmv**
This PR adds generic instrumentation to the main `callRestAPI`, `callSmartAPI`, and `callGraphQL` functions handling the API requests.

Screen Shot 2021-07-02 at 10 28 19 AM

### CODE REVIEW

**Comment by mstuart** [comment]
Looks good! But, for the event names, could we take the endpoint (Ex: `/foo/bar`) and include it in the event name like `foo_bar_error`? That'll allow us to do some quick event counts in our dashboards without having to peek into the "payload" part to figure out what API it was.

File: src/api/api.js [hunk 1]
```diff
@@ -37,11 +38,12 @@ export function callRestAPI<D, T>({ accessToken, method, url, data, headers } :
             const hasDetails = body.details && body.details.length;
             const issue = (hasDetails && body.details[0].issue) ? body.details[0].issue : 'Generic Error';
             const description = (hasDetails && body.details[0].description) ? body.details[0].description : 'no description';
-            
+
             const error = new Error(`${ issue }: ${ description } (Corr ID: ${ responseHeaders[HEADERS.PAYPAL_DEBUG_ID] }`);
             // $FlowFixMe
             error.response = { status, headers: responseHeaders };
 
+            getLogger().warn(`rest_api${ url.replace(/\//g, '_') }_error`, { err: issue });
```
Comment (mnicpt) [inline src/api/api.js:46]: @elizabethmv Wondering if we should create a helper function in logger.js in /lib to do `url.replace(/\//g, '_')`?

Comment (mnicpt) [reply src/api/api.js:46]: Maybe have a `formatSmartApiLog`, `fomatRestApiLog` and `formatGraphQlLog` that all pass in there specifics to a private generic one.

Comment (mnicpt) [reply src/api/api.js:46]: Actually remove `format` from those names.

Comment (elizabethmv) [reply src/api/api.js:46]: Yeah, I can pull this into a helper 👍 

It's just taking `/some/url` and doing `_some_url` to include it into the event name. Something like `slashToUnderscore(url)`?

Comment (elizabethmv) [reply src/api/api.js]: Updated! @mnicpt please re-review when you get a chance

**Comment by elizabethmv** [comment]
hi team, anyone free to review? thanks! @paypal/checkout-sdk

File: src/lib/logger.js [hunk 2]
```diff
@@ -93,3 +93,7 @@ export function setupLogger({ env, sessionID, clientID, sdkCorrelationID, buyerC
         logger.flush().catch(noop);
     });
 }
+
+export function slashToUnderscore(endpoint : string) : string {
```
Comment (gregjopa) [inline src/lib/logger.js:97]: Could we move this to `src/lib/util.js`? Seems more like a generic utility function.

**Review by gregjopa — APPROVED** [review]
Looks great! Nice work @elizabethmv 💯🥇👍

**Comment by mstuart** [comment]
lgtm!

**Review by mstuart — APPROVED** [review]

File: src/lib/logger.js (hunk 2 above)
Comment (elizabethmv) [reply src/lib/logger.js]: Great suggestion, updated 👍

**Pull request merged**